clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -1290,7 +1347,14 @@ invariant_checker_t::check_for_pc_discontinuity(
              shard->last_signal_context_.pre_signal_instr.memref.instr.addr) ||
 #endif
         // We expect a gap on a window transition.
-        shard->window_transition_ || prev_instr.instr.type == TRACE_TYPE_INSTR_SYSENTER;
+        shard->window_transition_ ||
+        prev_instr.instr.type ==
+            TRACE_TYPE_INSTR_SYSENTER IF_X86(
+                ||
+                (shard->between_kernel_syscall_trace_markers_ &&
+                 (shard->prev_instr_.decoding.opcode == OP_hlt ||
+                  // After interrupts are enabled, we may be interrupted.
+                  (shard->instrs_since_sti > 0 && shard->instrs_since_sti <= 2))));
```

Comment:
How did you compute 2?

Reply:
Empirically. It seemed to me that after enabling interrupts using STI, perhaps a queued interrupt is then delivered within 2 instrs.

Reply:
Added to comment.